src/v2/components/ecs-service/index.ts

```diff
@@ -0,0 +1,643 @@
+    tags?: pulumi.Input<EcsService.Tags>;
+  };
+}
+
+/**
+ * Standard directory permissions:
+ * - Owner: read, write, execute (7)
+ * - Group: read, execute (5)
+ * - Others: read, execute (5)
+ */
+const STANDARD_DIRECTORY_PERMISSIONS = '0755';
+
+const FIRST_POSIX_NON_ROOT_USER = {
+  userId: 1000,
+  groupId: 1000,
+  permissions: STANDARD_DIRECTORY_PERMISSIONS
+} as const;
+
+const defaults = {
+  desiredCount: 1,
+  size: 'small',
+  environment: [],
+  secrets: [],
+  volumes: [],
+  enableServiceAutoDiscovery: false,
+  assignPublicIp: false,
+  taskExecutionRoleInlinePolicies: [],
+  taskRoleInlinePolicies: [],
+  autoscaling: {
+    enabled: false,
+    minCount: 1,
+    maxCount: 1,
+  },
+};
+
+export class EcsService extends pulumi.ComponentResource {
+  name: string;
+  vpc: pulumi.Output<awsx.ec2.Vpc>;
+  logGroup: aws.cloudwatch.LogGroup;
+  taskDefinition: pulumi.Output<aws.ecs.TaskDefinition>;
+  taskExecutionRole: aws.iam.Role;
+  taskRole: aws.iam.Role;
+  service: aws.ecs.Service;
+  securityGroups: pulumi.Output<aws.ec2.SecurityGroup>[];
+  serviceDiscoveryService?: aws.servicediscovery.Service;
+  persistentStorage?: PersistentStorage;
+
+  constructor(
+    name: string,
+    args: EcsService.Args,
+    opts: pulumi.ComponentResourceOptions = {},
+  ) {
+    super('studion:ecs:Servicev2', name, {}, opts);
+    const argsWithDefaults = Object.assign({}, defaults, args);
+
+    this.name = name;
+    this.securityGroups = [];
+    this.vpc = pulumi.output(argsWithDefaults.vpc);
+    this.logGroup = this.createLogGroup();
+    this.taskExecutionRole = this.createTaskExecutionRole(argsWithDefaults.taskExecutionRoleInlinePolicies);
+    this.taskRole = this.createTaskRole(argsWithDefaults.taskRoleInlinePolicies)
+
+    if (argsWithDefaults.volumes.length) {
+      this.persistentStorage = this.createPersistentStorage(this.vpc);
+    }
+
+    this.taskDefinition = this.createTaskDefinition(
+      argsWithDefaults.containers,
+      argsWithDefaults.volumes,
+      this.taskExecutionRole,
+      this.taskRole,
+      argsWithDefaults.size,
+      { ...commonTags, ...argsWithDefaults.tags }
+    );
+
+    if (argsWithDefaults.enableServiceAutoDiscovery) {
+      this.serviceDiscoveryService = this.createServiceDiscovery();
+    }
+
+    this.service = this.createEcsService(argsWithDefaults);
+
+    if (argsWithDefaults.autoscaling.enabled) {
+      this.enableAutoscaling(
+        pulumi.output(argsWithDefaults.cluster).name,
+        this.service.name,
+        argsWithDefaults.autoscaling.minCount,
+        argsWithDefaults.autoscaling.maxCount,
+      );
+    }
+
+    this.registerOutputs();
+  }
+
+  public static createTcpPortMapping(port: number): aws.ecs.PortMapping {
+    return {
+      containerPort: port,
+      hostPort: port,
+      protocol: 'tcp'
+    };
+  }
+
+  private createLogGroup() {
+    const logGroup = new aws.cloudwatch.LogGroup(
+      `${this.name}-log-group`,
+      {
+        retentionInDays: 14,
```

Comment:
Why is this hardcoded? Shouldn't this be indefinite (which is set by default) for the production env?